### PR TITLE
Split first OOP / names / namespaces lecture in two

### PR DESCRIPTION
**Pull request by jstac**
This PR refactors the `Object Oriented Programming` lecture

### CODE REVIEW

**Comment by jstac** [comment]
@mmcky when you have a moment would you mind to look into this and tidy up?

File: lectures/oop_intro.md [hunk 1]
```diff
@@ -671,41 +670,33 @@ First,
...
 
 ```{figure} /_static/lecture_specific/oop_intro/global2.png
-:figclass: auto
 ```
 
 Next `g` is called via `y = g(10)`, leading to the following sequence of actions
 
 * The local namespace for the function is created.
 * Local names `x` and `a` are bound, so that the local namespace becomes `{'x': 10, 'a': 1}`.
+* Note that the global `a` was not affected by the local `a`.
```
Comment (mmcky) [inline lectures/oop_intro.md]: @jstac should this be indented? I think it belongs at the root level.

**Review by mmcky — COMMENTED** [review]
@jstac I noticed that the sections in `oop_intro`

- [ ]  `namespaces`
- [ ] `Viewing namespaces`
- [ ] `Interactive Sessions`
- [ ] `The global namespace`
- [ ] `Local Namespaces`
- [ ] `The _builtins_ namespace`
- [ ] `Name Resolution`

are pretty similar to `names` -- is this intention or should they be removed and make reference to the earlier lecture?

- [ ] I propose we add a `seealso` admonition in ``### {index}`Mutable <single: Mutable>` versus {index}`Immutable <single: Immutable>` parameters`` back to the `names` section on this topic.

**Comment by jstac** [comment]
Sorry @mmcky , I made a complete mess of this, doing it in a rush.

I've fixed the duplicates and the indent.

**Comment by jstac** [comment]
Will this spit out a new netlify version after my latest pushes are built?

**Comment by mmcky** [comment]
Thanks @jstac -- will review again tomorrow. 

Netlify has the latest. 

Should we switch the order for `OOP 1: Names and Namespaces` and `OOP 1: Objects and Methods`?

Screenshot 2024-03-01 at 9 48 48 pm

**Comment by jstac** [comment]
Hi @mmcky , thanks for picking that up.  I've fixed it.  Could you please look into why the build is failing?

**Comment by mmcky** [comment]
thanks @jstac this is now building and the preview is available here https://65e24dccd5105e2d2c5da1dc--epic-agnesi-957267.netlify.app/intro.html

I will try and proof read later today or tomorrow morning.

File: lectures/oop_intro.md [hunk 2]
```diff
@@ -295,7 +295,7 @@ inconsistent?
 The answers are related to the fact that Python aims for consistent style.
 
 In Python, it is common for users to build custom objects --- we discuss how to
-do this [later](python_oop).
+do this {doc}`lecture <python_oop>`.
```
Comment (jstac) [inline lectures/oop_intro.md]: @mmcky  Thanks for trying to fix but the sentence no longer makes sense.

Comment (jstac) [reply lectures/oop_intro.md]: I'll take care of this.

Comment (mmcky) [reply lectures/oop_intro.md]: thanks @jstac  -- apologies many of these are `in a later lecture` . Thanks for catching this.

**Comment by mmcky** [comment]
@jstac I will do a proof-read of the two lectures today.

**Pull request merged**